src/types/models/mod_feedback.rs

```diff
@@ -0,0 +1,157 @@
+    #[serde(skip_serializing)]
+    pub id: i32,
+    pub reviewer: Reviewer,
+    pub feedback_type: FeedbackTypeEnum,
+    pub feedback: String,
+    pub decision: bool,
+}
+
+#[derive(sqlx::Type, Serialize, Deserialize, Clone, PartialEq)]
+#[sqlx(type_name = "feedback_type")]
+pub enum FeedbackTypeEnum {
+    Positive,
+    Negative,
+    Suggestion,
+    Note
+}
+
+impl ModFeedback {
+    pub async fn get_for_mod_version_id(
+        version: &ModVersion,
+        note_only: bool,
+        pool: &mut PgConnection,
+    ) -> Result<ModFeedback, ApiError> {
+        let result = match sqlx::query_as!(
+            ModFeedbackRow,
+            r#"SELECT mf.id, mf.reviewer_id, dev.display_name AS reviewer_name, dev.admin AS reviewer_admin, mf.type AS "feedback_type: _", mf.feedback, mf.decision
+            FROM mod_feedback mf
+			INNER JOIN developers dev ON dev.id = mf.reviewer_id
+            WHERE mf.mod_version_id = $1"#,
+            version.id
+        )
+        .fetch_all(&mut *pool)
+        .await
+        {
+            Err(e) => {
+                log::error!("{}", e);
+                return Err(ApiError::DbError);
+            }
+            Ok(r) => r,
+        };
+
+        let feedback: Vec<ModFeedbackOne> = result.into_iter().filter_map(|row| {
+            if note_only && row.feedback_type != FeedbackTypeEnum::Note {
+                None
+            } else {
+                Some(ModFeedbackOne {
+                    id: row.id,
+                    reviewer: Reviewer {
+                        id: row.reviewer_id,
+                        display_name: row.reviewer_name,
+                        admin: row.reviewer_admin,
+                    },
+                    feedback_type: row.feedback_type,
+                    feedback: row.feedback,
+                    decision: row.decision,
+                })
+            }
+        }).collect();
+
+        let positive = feedback.iter().filter(|r| r.feedback_type == FeedbackTypeEnum::Positive).count() as i32;
+        let negative = feedback.iter().filter(|r| r.feedback_type == FeedbackTypeEnum::Negative).count() as i32;
+        let return_res =
+            ModFeedback {
+                score: positive - negative,
+                mod_id: version.mod_id.clone(),
+                mod_version: version.version.clone(),
+                feedback,
+            };
+
+        Ok(return_res)
+    }
+
+    pub async fn set(
+        version: &ModVersion,
+        reviewer_id: i32,
+        feedback_type: FeedbackTypeEnum,
+        feedback: &str,
+        decision: bool,
+        pool: &mut PgConnection
+    ) -> Result<(), ApiError> {
+        sqlx::query!(
+            r#"INSERT INTO mod_feedback (mod_version_id, reviewer_id, type, feedback, decision)
+            VALUES ($1, $2, $3, $4, $5)
+            ON CONFLICT (mod_version_id, reviewer_id)
+            DO UPDATE SET type = EXCLUDED.type, feedback = EXCLUDED.feedback, decision = EXCLUDED.decision"#,
```

Comment:
we're limiting mods to only one review per developer? why

Reply:
it would be a lot more annoying to edit - how would you know which one you're editing???

Reply:
i think i'd rather disallow editing and allow devs to post multiple feedback (like if they wanted to add a few notes and a suggestion) - overwriting old feedback isn't very obvious behavior and may be confusing if the dev responds to some bit of feedback and then the old one is edited. could add a patch for feedback with id as well, but not really necessary

Reply:
yeah, i'll remove updating for now and add multi-posting.